src/overlay/Peer.cpp

```diff
@@ -778,7 +776,11 @@ Peer::recvSCPMessage(StellarMessage const& msg)
                                 ? mRecvSCPExternalizeTimer.TimeScope()
                                 : (mRecvSCPNominateTimer.TimeScope()))));
 
-    mApp.getHerder().recvSCPEnvelope(envelope);
+    auto res = mApp.getHerder().recvSCPEnvelope(envelope);
+    if (res != Herder::ENVELOPE_STATUS_DISCARDED)
```

Comment:
I'm a bit unclear on this. You changed the recvSCPEnvelope path to return ENVELOPE_STATUS_SKIPPED_SELF in the previous change, and I would have thought that SKIPPED_SELF should also imply _not_ flooding the message further, but in this condition it looks like a SKIPPED_SELF message _will_ get flooded further. Can you explain a bit what is being inhibited here, and why it's correct?

Reply:
SKIPPED_SELF just means that it was an SCP message created by the local node, so we do want to flood it; the meaning of ENVELOPE_STATUS_DISCARDED now is that it's a message that is discarded because it's not in the right range of ledgers regardless of where it came from.